Simplified/Book/UI/NYPLBookCellDelegate+Audiobooks.m

```diff
@@ -216,6 +223,34 @@ - (void)scheduleTimerForAudiobook:(NYPLBook *)book
   }];
 }
 
+// Create a timer that saves the audiobook progress periodically when the app is inactive.
+// We do save the progress when app is being killed and applicationWillTerminate: is called,
+// but applicationWillTerminate: is not always called when users force quit the app.
+// This method is triggered when app resigns active.
+- (void)scheduleAudiobookProgressSavingTimer {
+  if (self.audiobookProgressSavingTimer) {
+    return;
+  }
+  
+  self.audiobookProgressSavingTimer = [NSTimer scheduledTimerWithTimeInterval:AudiobookProgressSavingInterval
+                                                                      repeats:YES
+                                                                        block:^(NSTimer *_Nonnull timer) {
+    [NYPLMainThreadRun asyncIfNeeded:^{
+      if ([[UIApplication sharedApplication] applicationState] == UIApplicationStateActive) {
+        [timer invalidate];
```

Comment:
it seems like the only way to stop / invalidate the timer is here, that is when the app becomes active again. But what it doesn't become active again, and the user simply  stops playback from the device lock screen (without killing the app) ?  Will this loop continue to run forever ?

Reply:
The timer is now suspended if user stops playback from lock screen. It will also be released if it's owner (AudiobookManager) is released from memory.